### PR TITLE
#2695 - Eth withdraw redesign

### PR DESCRIPTION
**Pull request by drakkein**

### CODE REVIEW

File: app/components/modals/tx-sender/withdraw-flow/Withdraw.unsafe.tsx [hunk 1]
```diff
@@ -76,84 +110,214 @@ const getWithdrawFormSchema = (maxEther: string) =>
...
+            </Button>
+          </section>
         </Form>
       )}
     </Formik>
   </section>
 );
 
-const Withdraw = compose<TProps & IIntlProps, {}>(
+const WithdrawUnsafe = compose<TProps, {}>(
```
Comment (Moejoe90) [inline app/components/modals/tx-sender/withdraw-flow/Withdraw.unsafe.tsx]: why do you call it `unsafe`?

File: app/modules/sagasUtils.ts [hunk 2]
```diff
@@ -123,3 +124,16 @@ export function* neuRestartIf(
     });
   }
 }
+
+/**
+ * Ignore incoming actions for a given period of time while processing a task.
+ * After spawning a task, it's still accepting incoming actions into the underlying buffer, keeping at most 1.
+ */
+export function* neuThrottle(
```
Comment (Moejoe90) [inline app/modules/sagasUtils.ts]: we need unit tests for this

File: app/modules/tx/sender/actions.ts [hunk 3]
```diff
@@ -36,4 +36,8 @@ export const txSenderActions = {
   // reducer setters
   setTransactionData: (txData?: ITxData) =>
     createAction("TX_SENDER_SET_TRANSACTION_DATA", { txData }),
+  setAdditionalData: createActionFactory(
+    "TX_SENDER_SET_ADDITIONAL_DATA",
+    (additionalData: any) => ({ additionalData }),
```
Comment (Moejoe90) [inline app/modules/tx/sender/actions.ts]: I would say for` withdraw` we know all the additional data correct?

Let's do it like this each transaction has its own clear additional data and then additional data can be 
`additionalData: IWithdrawAdditionalData | IUpgradeAdditionalData` etc

File: app/components/shared/MoneySuiteWidget.tsx [hunk 4]
```diff
@@ -34,28 +42,40 @@ export const MoneySuiteWidget: React.FunctionComponent<IMoneySuiteWidgetProps &
...
-      <div className={styles.money} data-test-id={makeTid(dataTestId, "large-value")}>
+      <div
+        className={cn(styles.money, "text-right")}
+        data-test-id={makeTid(dataTestId, "large-value")}
+      >
         <MoneyNew
           value={largeNumber}
           inputFormat={ENumberInputFormat.ULPS}
-          outputFormat={ENumberOutputFormat.ONLY_NONZERO_DECIMALS}
+          outputFormat={outputFormat!}
```
Comment (Moejoe90) [inline app/components/shared/MoneySuiteWidget.tsx]: this looks like code smell 
`outputFormat` in this case shouldn't be optional

File: app/components/shared/MoneySuiteWidget.tsx [hunk 5]
```diff
@@ -68,3 +88,7 @@ export const MoneySuiteWidget: React.FunctionComponent<IMoneySuiteWidgetProps &
     </div>
   </div>
 );
+
+MoneySuiteWidget.defaultProps = {
+  outputFormat: ENumberOutputFormat.ONLY_NONZERO_DECIMALS,
```
Comment (Moejoe90) [inline app/components/shared/MoneySuiteWidget.tsx]: please define this up, make `outputFormat` with a default value and remove `!`

File: app/modules/tx/validator/sagas.ts [hunk 6]
```diff
@@ -35,8 +39,28 @@ export function* txValidateSaga({ logger }: TGlobalDependencies, action: TAction
 
   try {
     generatedTxDetails = yield neuCall(validationGenerator, action.payload);
+
+    // Process additional data for withdrawal transaction
+    if (action.payload.type === ETxSenderType.WITHDRAW) {
```
Comment (Moejoe90) [inline app/modules/tx/validator/sagas.ts]: please put this in a separate saga and yield from here similar to `validateGas`

File: app/test-e2e/wallets/WalletWithdraw.spec.e2e.ts [hunk 7]
```diff
@@ -114,7 +123,7 @@ describe("Wallet Withdraw", () => {
         // cy.get(tid("modals.tx-sender.withdraw-flow.summary.cost"))
         // .then(e => parseAmount(e.text()).toNumber())
         // .should("be.closeTo", 0.0002, 0.0001);
-        cy.get(tid("modals.tx-sender.withdraw-flow.summary.cost")).contains(/0\.\d{4}/);
+        cy.get(tid("modals.tx-sender.withdraw-flow.summary.cost.large-value")).contains(/0\.\d{4}/);
```
Comment (Moejoe90) [inline app/test-e2e/wallets/WalletWithdraw.spec.e2e.ts]: please add to the e2e the flow where user enters 
1 - a smart contract 
2 - user enters an empyty unused wallet

File: app/components/modals/tx-sender/withdraw-flow/WithdrawTransactionDetails.tsx [hunk 8]
```diff
@@ -1,58 +1,98 @@
...
+        </EtherscanAddressLink>
+      </div>
+    </section>
+
+    <section className={cn(styles.section, "flex-nowrap")}>
+      <FormattedMessage id="modal.sent-eth.to-address" />
+      <div className={cn(styles.withEtherscan, "text-right")}>
+        <span
+          className={cn(
+            styles.walletLabel,
```
Comment (Moejoe90) [inline app/components/modals/tx-sender/withdraw-flow/WithdrawTransactionDetails.tsx]: this looks like it needs its own style  in `Withdraw.module.scss`

File: app/components/modals/tx-sender/shared/TxPending.stories.tsx [hunk 9]
```diff
@@ -2,37 +2,56 @@ import { action } from "@storybook/addon-actions";
...
-import { ETxSenderType } from "../../../../modules/tx/types";
+import { ETokenType, ETxSenderType } from "../../../../modules/tx/types";
 import { withModalBody } from "../../../../utils/storybookHelpers.unsafe";
 import { TxPendingLayout } from "./TxPending";
 
 const txData: React.ComponentProps<typeof TxPendingLayout> = {
   blockId: 4623487932,
   txHash: "af908098b968d7564564362c51836",
-  type: ETxSenderType.WITHDRAW,
+  type: ETxSenderType.UPGRADE,
```
[the remark this answers is not in the document]
Comment (drakkein) [reply app/components/modals/tx-sender/shared/TxPending.stories.tsx]: Withdraw got different PENDING design, so i moved it to second story called "simple"

File: app/components/modals/tx-sender/TxSender.tsx [hunk 10]
```diff
@@ -36,7 +36,7 @@ import { UserClaimSuccess } from "./user-claim/Success";
 import { UserClaimSummary } from "./user-claim/Summary";
 import { WithdrawSuccess } from "./withdraw-flow/Success";
 import { WithdrawSummary } from "./withdraw-flow/Summary";
-import { Withdraw } from "./withdraw-flow/Withdraw.unsafe";
+import { WithdrawUnsafe } from "./withdraw-flow/Withdraw.unsafe";
```
[the remark this answers is not in the document]
Comment (drakkein) [reply app/components/modals/tx-sender/TxSender.tsx]: Ahh... My editor renamed everything automatically. I've reverted this change

File: app/components/modals/tx-sender/withdraw-flow/Withdraw.stories.tsx [hunk 11]
```diff
@@ -1,17 +1,26 @@
...
+      gasPrice={convertToBigInt(0.002)}
+      gasPriceEur={convertToBigInt(0.2)}
+      total={convertToBigInt(124124.002)}
+      totalEur={convertToBigInt(124124.2)}
+      walletAddress="0x327t3278gf27c82bsimfoc2"
+      onAccept={() => action("onAccept")}
+      additionalData={{}}
       maxEther={"100000000000000000000000000000"}
       onValidateHandler={() => {}}
-      intl={dummyIntl}
```
[the remark this answers is not in the document]
Comment (drakkein) [reply app/components/modals/tx-sender/withdraw-flow/Withdraw.stories.tsx]: there are no more dynamic translations, all are done using FormattedMessage component

File: app/test-e2e/wallets/WalletWithdraw.spec.e2e.ts [hunk 12]
```diff
@@ -24,9 +24,11 @@ const typeWrongAddress = () => {
 const typeWrongValue = () => {
   assertWithdrawButtonIsDisabled();
   cy.get(tid("modals.tx-sender.withdraw-flow.withdraw-component.value")).type("-1");
-  assertWithdrawButtonIsDisabled();
+  cy.get(tid("modals.tx-sender.withdraw-flow.withdraw-component.value")).should("have.value", "1");
+  cy.get(tid("modals.tx-sender.withdraw-flow.withdraw-component.value")).clear();
+
```
[the remark this answers is not in the document]
Comment (drakkein) [reply app/test-e2e/wallets/WalletWithdraw.spec.e2e.ts]: We've got masked input used right now, so even if you try to write not allowed value, all non-valid characters will be ignored.

File: app/components/modals/tx-sender/withdraw-flow/Withdraw.unsafe.tsx [hunk 13]
```diff
@@ -76,84 +110,214 @@ const getWithdrawFormSchema = (maxEther: string) =>
...
+          </section>
+
+          <section className="mb-4">
+            <FormLabel for="to" className={styles.label}>
+              <FormattedMessage id="modal.sent-eth.to-address" />
+            </FormLabel>
+            <FormInput
+              name="to"
+              data-test-id="modals.tx-sender.withdraw-flow.withdraw-component.to-address"
+              maxLength={42}
```
[the remark this answers is not in the document]
Comment (drakkein) [reply app/components/modals/tx-sender/withdraw-flow/Withdraw.unsafe.tsx]: `maxLength` only a limiter to the input length while `charactersLimit` also adds characters counter under input

File: app/components/modals/tx-sender/withdraw-flow/Withdraw.unsafe.tsx [hunk 14]
```diff
@@ -76,84 +110,214 @@ const getWithdrawFormSchema = (maxEther: string) =>
...
+              size={InputSize.SMALL}
+            />
+            <Button
+              data-test-id="bank-transfer.reedem-init.redeem-whole-balance"
+              className={"mt-2"}
+              onClick={() => {
+                setFieldValue(
+                  "value",
+                  getFormattedMoney(
+                    maxEther,
```
[the remark this answers is not in the document]
Comment (drakkein) [reply app/components/modals/tx-sender/withdraw-flow/Withdraw.unsafe.tsx]: yes

File: app/test-e2e/wallets/WalletWithdraw.spec.e2e.ts (hunk 7 above)
Comment (drakkein) [reply app/test-e2e/wallets/WalletWithdraw.spec.e2e.ts]: Do you have any idea what should be source of truth for smart contract address?

Comment (Moejoe90) [reply app/test-e2e/wallets/WalletWithdraw.spec.e2e.ts]: We already discussed this but please use one of the fixture addresses 
There was also a problem with addresses that do not accept ether so make sure you use a smart contract that accepts ether

File: app/components/modals/tx-sender/withdraw-flow/Withdraw.unsafe.tsx [hunk 15]
```diff
@@ -76,84 +110,214 @@ const getWithdrawFormSchema = (maxEther: string) =>
...
+              <FormattedMessage id="modal.sent-eth.to-address" />
+            </FormLabel>
+            <FormInput
+              name="to"
+              data-test-id="modals.tx-sender.withdraw-flow.withdraw-component.to-address"
+              maxLength={42}
+              charactersLimit={42}
+              size={InputSize.SMALL}
+            />
+            <EtherscanAddressLink
```
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/modals/tx-sender/withdraw-flow/Withdraw.unsafe.tsx]: DId you create a ticket for this?

[the remark this answers is not in the document]
Comment (drakkein) [reply app/components/modals/tx-sender/withdraw-flow/Withdraw.unsafe.tsx]: https://github.com/Neufund/platform-frontend/issues/3123

File: app/test-e2e/wallets/WalletWithdraw.spec.e2e.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/test-e2e/wallets/WalletWithdraw.spec.e2e.ts]: But this is done after `clear()` so there is no input to mask in the first place

**Pull request merged**